lib/IR/LegacyPassManager.cpp

```diff
@@ -1883,6 +1883,7 @@ void FunctionPass::assignPassManager(PMStack &PMS,
 
     // [1] Create new Function Pass Manager
     FPP = new FPPassManager();
+    std::unique_ptr<FPPassManager> NewFPP(FPP); // HLSL Change
```

Comment:
This was found through the OOM test. Presumably there is a `SmallVector` in one of the calls below that will now throw, such that the `FPPassManager` would leak. The lifetime of objects here is really ugly and obscure :/